profiles/management/commands/createdefaultsuperuser.py

```
@@ -0,0 +1,29 @@
+import os
+from portal.settings import (
+    CREATE_DEFAULT_SU,
+    SU_DEFAULT_PASSWORD,
+    AUTH_PASSWORD_VALIDATORS,
+)
+from django.contrib.auth.password_validation import validate_password
+from django.core.management.base import BaseCommand, CommandError
+from profiles.models import HealthcareUser
+
+SU_EXISTS = HealthcareUser.objects.filter(name="hcw_portal_admin").exists()
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        if CREATE_DEFAULT_SU and SU_DEFAULT_PASSWORD and not SU_EXISTS:
+            self.create_su()
+        elif SU_EXISTS:
+            self.stdout.write(self.style.SUCCESS("Default Super User already exists"))
+        else:
+            self.stdout.write(self.style.SUCCESS("Not creating default Super User"))
+
+    def create_su(self):
+        ## Check to ensure SU password meets minimum requirements
+        if validate_password(SU_DEFAULT_PASSWORD) is None:
```

There's no message for users if the password doesn't work. It seems like that would make it fail silently.

---

The Validation Error returned by the validate_password is raised and displays an error message in stdout with why the password didn't pass validation.  I could get fancy and catch the errors and then echo them out.  Would still need to throw a validation error though otherwise the test/build will think it passed.

---

Ah, all good then.